### PR TITLE
Enabled helpful NullPointerException if jdk >= 14

### PR DESCRIPTION
**Pull request by astrelsky**
I tested the following in Windows and using wsl for linux. Maybe this would prove useful once in a blue moon.

### CODE REVIEW

File: Ghidra/RuntimeScripts/Linux/support/launch.sh [hunk 1]
```diff
@@ -123,6 +123,11 @@ if [ "${MAXMEM}" != "" ]; then
 	VMARG_LIST+=" -Xmx${MAXMEM}"
 fi
 
+# if using jdk 14 or later enable helpful NullPointerException
+if [ $(java -version 2>&1 | sed -E -n 's/.* version "([^.-]*).*"/\1/p' | cut -d' ' -f1) -ge 14 ]; then
```
Comment (ryanmkurtz) [inline Ghidra/RuntimeScripts/Linux/support/launch.sh:127]: You need to be checking the version of $JAVA_CMD.  That will ultimately be the java that Ghidra ends up using.

**Comment by ryanmkurtz** [comment]
A much simpler alternative is to just throw this new option in the `Ghidra.launch` file.  `-XX:+IgnoreUnrecognizedVMOptions` is already defined there so you don't have to worry about checking a Java version.  Also, this feature would be much more important to developers, and they are most likely running out of Eclipse anyway.

**Comment by astrelsky** [comment]
> A much simpler alternative is to just throw this new option in the `Ghidra.launch` file.  `-XX:+IgnoreUnrecognizedVMOptions` is already defined there so you don't have to worry about checking a Java version.  Also, this feature would be much more important to developers, and they are most likely running out of Eclipse anyway.

I see.  I will most likely take that approach. I am unfamiliar with the Java vm options and didn't know there's a convenient option to ignore unrecognized ones.

The reason I decided to submit this is because it can prove helpful to developers when incomplete issues are submitted. There have been times where issues have been submitted with the template left unmodified and the only additions being a stack trace and something along the lines of "ghidra broke". I am assuming these are due to a language barrier which only so much can be done about. In the event of a NullPointerException if the user is using a jdk > 14 the dev team will at least be able to know what variable was null without the daunting task of reproducing it without having enough information.

**Comment by ryanmkurtz** [comment]
I do think it's a cool feature and will add value to the stack traces.  I was just offering a simpler alternative to make the PR a "no-brainer" to accept.  Modifying those launch scripts requires careful testing on all 3 platforms.  `-XX:+IgnoreUnrecognizedVMOptions` and the new property could also be added to `launch.properties` directly as yet another alternative.

File: Ghidra/RuntimeScripts/Linux/support/launch.sh (hunk 1 above)
Comment (astrelsky) [reply Ghidra/RuntimeScripts/Linux/support/launch.sh]: I have modified the changes to instead append both `XX:+IgnoreUnrecognizedVMOptions` and `-XX:+ShowCodeDetailsInExceptionMessages` to the vmargs. Should the vmargs be checked first for an unrecognized option from the user before appending these?

File: Ghidra/RuntimeScripts/Linux/support/launch.sh [hunk 2]
```diff
@@ -154,6 +154,8 @@ else
 	exit 1
 fi
 
+VMARG_LIST+=" -XX:+IgnoreUnrecognizedVMOptions -XX:+ShowCodeDetailsInExceptionMessages"
```
Comment (ryanmkurtz) [inline Ghidra/RuntimeScripts/Linux/support/launch.sh:157]: Do you know about the launch.properties file?

Comment (astrelsky) [reply Ghidra/RuntimeScripts/Linux/support/launch.sh:157]: Yes. For some reason it didn't click to use it but that should have been obvious.

Comment (ryanmkurtz) [reply Ghidra/RuntimeScripts/Linux/support/launch.sh:157]: Unfortunately the Ghidra.launch file still needs to be updated.  I really hoped that there was a JVM argument that could take a path to a properties file so they wouldn't have to be defined twice, but I couldn't find anything like that.

Comment (astrelsky) [reply Ghidra/RuntimeScripts/Linux/support/launch.sh]: > Unfortunately the Ghidra.launch file still needs to be updated.  I really hoped that there was a JVM argument that could take a path to a properties file so they wouldn't have to be defined twice, but I couldn't find anything like that.

I've set them in both. I think this should be good now. My only concern is that if a user adds a vmarg and makes a typo they may not realize it is being ignored which could be a headache to debug.

**Comment by ryanmkurtz** [comment]
This has been turned on by default in Java 15 and up.

**Pull request closed without merging**